back/app/models/initiative.rb

```diff
@@ -110,6 +110,20 @@ class Initiative < ApplicationRecord
   scope :no_feedback_needed, -> { with_status_code(InitiativeStatus::CODES - ['threshold_reached']) }
   scope :proposed, -> { with_status_code('proposed') }
 
+  def cosponsor_ids=(ids)
+    return unless ids
+
+    ids = ids.uniq
+    current_ids = cosponsors.pluck(:id).uniq
+    return if current_ids.sort == ids.sort
+
+    cosponsors_initiatives.where.not(user_id: ids).destroy_all
+
+    (ids - current_ids).each do |id|
+      cosponsors_initiatives.create(user_id: id)
+    end
+  end
```

Comment:
```suggestion
  def cosponsor_ids=(ids)
    return unless ids
    super(ids.uniq)
  end
```

Sorry I didn't think about it before, but it seems the implementation can be significantly simplified. At least the current tests pass. And maybe we can also remove `return unless ids`.

Reply:
Yes, that is significantly simpler!